### ISSUE TITLE
1장 IMDB를 GRU로 분석하는 코드에서

### ISSUE THREAD

**Issue by metamath1** [ai-agent-kr/Modern-AI-Agents#1]
```
class SentimentRNN(nn.Module):
    def __init__(self, no_layers, vocab_size, hidden_dim, embedding_dim, drop_prob=0.5):
        super(SentimentRNN, self).__init__()
        self.output_dim = output_dim
        self.hidden_dim = hidden_dim
        self.no_layers = no_layers
        self.vocab_size = vocab_size

        self.embedding = nn.Embedding(vocab_size, embedding_dim)
        self.rnn = nn.GRU(input_size=embedding_dim, hidden_size=self.hidden_dim,
                          num_layers=no_layers, batch_first=True)
        self.dropout = nn.Dropout(drop_prob)
        self.fc = nn.Linear(self.hidden_dim, output_dim)
        self.sig = nn.Sigmoid()

    def forward(self, x, hidden):
        batch_size = x.size(0)
        embeds = self.embedding(x)
        rnn_out, hidden = self.rnn(embeds, hidden)
        rnn_out = rnn_out.contiguous().view(-1, self.hidden_dim)
        out = self.dropout(rnn_out)
        sig_out = self.sig(out)
        sig_out = sig_out.view(batch_size, -1)
        sig_out = sig_out[:, -1]
        return sig_out, hidden

    def init_hidden(self, batch_size):
        ''' Initializes hidden state '''

        h0 = torch.zeros((self.no_layers, batch_size, self.hidden_dim)).to(device)
        return h0
```

위 코드에서 포워드할 때 self.fc 를 사용하지 않았습니다. 책 본문에는 사용한다고 되어있습니다.
책설명은  드롭아웃후 선형 레이어를 사용하는것처럼 되어 있습니다.

이 코드는 결과적으로 마지막 텐서가 (샘플수, 타임스탭*히든사이즈) 이라 
last hidden 벡터의 인덱스 255번 숫자를 뽑아서 학습에 
사용하는 방식으로 되어 있는데 이런 식이면 dropout 레이어를 사용할 의미가 없을것 같은데요.
원 저자가 왜 이렇게 코딩을 했는지.... 저는 첨보는 방식인데요.
의도가 없다면 오류같은데 별도의 설명이 없다면 책을 꼼꼼하게 읽는 입문자들에게 매우 혼란스런 부분인것같습니다.
혹시 이 부분에 다른 의도가 있는건가요?
